File: src/main/java/org/code4everything/hutool/converter/PatternConverter.java
package org.code4everything.hutool.converter;

import cn.hutool.core.util.StrUtil;
import org.code4everything.hutool.Converter;

import java.util.regex.Pattern;

/**
 * @author pantao
 * @since 2020/10/31
 */
public class PatternConverter implements Converter<Pattern> {

    @Override
    public Pattern string2Object(String string) {
        return Pattern.compile(string);
    }

    @Override
    public String object2String(Object object) {
        return object instanceof Pattern ? ((Pattern) object).pattern() : StrUtil.EMPTY;
    }
}

File: src/main/kotlin/org/code4everything/hutool/converter/PatternConverter.kt
package org.code4everything.hutool.converter

import java.util.regex.Pattern
import org.code4everything.hutool.Converter

class PatternConverter : Converter<Pattern> {

    override fun string2Object(string: String): Pattern = Pattern.compile(string)

    override fun object2String(any: Any?): String = if (any is Pattern) any.pattern() else ""
}
